crates/ruff_python_formatter/src/lib.rs

```diff
@@ -57,70 +55,106 @@ pub fn fmt(contents: &str) -> Result<Formatted<ASTFormatContext>> {
 
 #[cfg(test)]
 mod tests {
+    use std::fs;
     use std::path::Path;
 
     use anyhow::Result;
-    use test_case::test_case;
 
     use crate::fmt;
-    use crate::test::test_resource_path;
-
-    #[test_case(Path::new("simple_cases/beginning_backslash.py"); "beginning_backslash")]
-    #[test_case(Path::new("simple_cases/class_blank_parentheses.py"); "class_blank_parentheses")]
-    #[test_case(Path::new("simple_cases/class_methods_new_line.py"); "class_methods_new_line")]
-    #[test_case(Path::new("simple_cases/import_spacing.py"); "import_spacing")]
-    #[test_case(Path::new("simple_cases/one_element_subscript.py"); "one_element_subscript")]
-    #[test_case(Path::new("simple_cases/power_op_spacing.py"); "power_op_spacing")]
-    #[test_case(Path::new("simple_cases/remove_newline_after_code_block_open.py"); "remove_newline_after_code_block_open")]
-    #[test_case(Path::new("simple_cases/slices.py"); "slices")]
-    #[test_case(Path::new("simple_cases/tricky_unicode_symbols.py"); "tricky_unicode_symbols")]
-    // Passing except that `1, 2, 3,` should be `(1, 2, 3)`.
-    #[test_case(Path::new("simple_cases/tupleassign.py"); "tupleassign")]
-    // Passing except that `CliRunner().invoke(...)` arguments are improperly wrapped.
-    #[test_case(Path::new("simple_cases/function2.py"); "function2")]
-    fn passing(path: &Path) -> Result<()> {
-        let snapshot = format!("{}", path.display());
-        let content = std::fs::read_to_string(test_resource_path(
-            Path::new("fixtures/black").join(path).as_path(),
-        ))?;
+    use ruff_testing_macros::fixture;
+    use similar::TextDiff;
+    use std::fmt::{Formatter, Write};
+
+    #[fixture(
+        pattern = "resources/test/fixtures/black/**/*.py",
+        // Excluded tests because they reach unreachable when attaching tokens
+        exclude = [
+            "*comments.py",
+            "*comments[3,5,8].py",
+            "*comments_non_breaking_space.py",
+            "*docstring_preview.py",
+            "*docstring.py",
+            "*fmtonoff.py",
+            "*fmtskip8.py",
+            "*function2.py",
+            "*one_element_subscript.py",
+            "*remove_except_parens.py",
+            "*remove_parens.py",
+            "*remove_newline_after_code_block_open.py",
+            "*slices.py",
+            "*string_prefixes.py",
+            "*tricky_unicode_symbols.py",
+            "*tupleassign.py"
```

Comment:
There are now more tests that panic than when I initially created this PR. I think it would be good for us to fix the panics first to get better visibility of our changes (and avoid having any new tests that cause panics)